src/unix/syscall.c

```diff
@@ -2476,6 +2477,8 @@ void syscall_debug(context f)
     // i dont know that we actually want to defer the syscall return...its just easier for the moment to hew
     // to the general model and make exceptions later
     schedule_frame(f);
+  out:
+    kern_unlock();
```

Comment:
Okay, this might be a bit too nitpicky, but it might be nice to keep lock and unlock together, either move kern_try_lock in here, or else move unlock/runloop in syscall_schedule, and let it call runloop exclusively.

Reply:
In general I would agree about keeping lock acquire/release paired, but this isn't really possible when a scheduled thunk is invoked with the kernel lock already acquired (e.g. resume_syscall). For that matter, anything that is served off of the runqueue (or syscall handler) is invoked with the lock held, so a subsequent call to runloop (excepting kernel fault case) will require an (unpaired) unlock.